src/luxonis_ml/data/dataset_recognition.py

```diff
@@ -27,37 +27,74 @@ def recognize(dataset_path: str) -> str:
     yaml_files = list(
         filter(lambda file_dir: file_dir.split(".")[-1] == "yaml", file_dir_list)
     )
+    csv_files = list(
+        filter(lambda file_dir: file_dir.split(".")[-1] == "csv", file_dir_list)
+    )
+    npy_files = list(
+        filter(lambda file_dir: file_dir.split(".")[-1] == "npy", file_dir_list)
+    )
+    txt_files = list(
+        filter(lambda file_dir: file_dir.split(".")[-1] == "txt", file_dir_list)
+    )
+    xml_files = list(
+        filter(lambda file_dir: file_dir.split(".")[-1] == "xml", file_dir_list)
+    )
     dirs = list(filter(lambda file_dir: os.path.isdir(file_dir), file_dir_list))
 
+    # COCO Dataset
     if json_files and dirs:
-        if len(json_files) > 1:
-            raise Exception(
-                "Possible COCO dataset but multiple json files present - possible ambiguity."
-            )
-        json_file = json_files[0]
-
-        try:
-            with open(json_file) as file:
-                json.load(file)
-        except ValueError:
-            raise Exception(f"{json_file} is not a valid json file.")
-
-        # NOTE: if we want to validate the file further, we can use `if all(key in coco for key in ['images', 'annotations', 'categories']):``
-        return "COCO"
-
-    if yaml_files and dirs:
-        if len(yaml_files) > 1:
-            raise Exception(
-                "Possible YOLO dataset but multiple yaml files present - possible ambiguity."
-            )
-        yaml_file = yaml_files[0]
-        try:
-            yaml.safe_load(Path(yaml_file).read_text())
-        except ScannerError:
-            raise Exception(f"{yaml_file} is not a valid yaml file.")
-
-        return "YAML"
-
-    return "Unknown dataset"
-    # TODO: - rest of datasets + appropriate parsers call
-    ...
+        with open(json_files[0], 'r') as f:
+            data = json.load(f)
+        if 'images' in data and 'annotations' in data and 'categories' in data:
+            return "COCO"
```

Comment:
Let's turn those into enums?
DatasetFormat? Or similar? CC @conorsim

Reply:
@tersekmatija you mean the return value right? Yeah, I think this could just be a `DatasetFormat` enum.

Reply:
We do have [this enum](https://github.com/luxonis/luxonis-ml/blob/e73d9dd5c40c67d651a2d8b3e58febfee43ff812/src/luxonis_ml/data/parsers.py#L21) already it looks like. Maybe we can generalize it to apply to both `dataset_recognition.py` and `parsers.py`